src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -127,13 +127,14 @@ private bool StripAssembly(ITaskItem assemblyItem)
 
         try
         {
-            if(!AssemblyStripper.AssemblyStripper.TryStripAssembly(assemblyFile, outputPath))
+            if (!AssemblyStripper.AssemblyStripper.TryStripAssembly(assemblyFile, outputPath))
             {
                 Log.LogMessage(MessageImportance.Low, $"[ILStrip] Skipping {assemblyFile} because it is not a managed assembly.");
             }
             else
             {
-                _processedAssemblies.GetOrAdd(assemblyItem.ItemSpec, GetTrimmedAssemblyItem(assemblyItem, outputPath, assemblyFile));
+                var fullPath = assemblyItem.GetMetadata("FullPath");
```

Comment:
Is `FullPath` always available as assembly item's metadata?

Reply:
Yes, `FullPath` is one of the built-in / well known metadata which is always available: https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-well-known-item-metadata?view=vs-2022

Reply:
My understanding is that MSBuild assigns this metedata for us.

Reply:
Ah, thanks for the clarification